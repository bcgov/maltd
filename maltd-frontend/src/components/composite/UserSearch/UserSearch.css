body {
  color: #303030 !important;
}

.error-message {
  color: #dc3545;
}

.move-down {
  margin-top: 30px;
}

.error-btn-div {
  display: flex;
  justify-content: space-between;
}

@media only screen and (max-width: 600px) {
  .error-btn-div {
    line-height: 1;
  }
}
