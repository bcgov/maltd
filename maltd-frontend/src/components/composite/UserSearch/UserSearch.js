/* eslint-disable react/jsx-filename-extension */
import React from "react";
import PropTypes from "prop-types";
import InputField from "../../base/InputField/InputField";
import GeneralButton from "../../base/GeneralButton/GeneralButton";
import "./UserSearch.css";

export default function UserSearch({
  userSearch: { state },
  inputField,
  generalButton
}) {
  return (
    <div>
      <p>
        Find a user by IDIR username and add them to or remove them from
        projects.
      </p>
      <p />
      {!state.isLoading && (
        <div>
          <InputField inputField={inputField} />
          <p />
          <GeneralButton generalButton={generalButton} />
        </div>
      )}
      {state.isLoading && (
        <div>
          <div>
            <InputField inputField={inputField} />
            <p />
            <div>
              <div className="float-left">
                <p className="float-left">
                  <small>Loading...</small>
                </p>
              </div>
              <GeneralButton generalButton={generalButton} />
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

UserSearch.propTypes = {
  userSearch: PropTypes.shape({
    state: PropTypes.any.isRequired
  }).isRequired,
  inputField: PropTypes.shape({
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    valid: PropTypes.bool.isRequired,
    invalid: PropTypes.bool.isRequired,
    value: PropTypes.string.isRequired
  }).isRequired,
  generalButton: PropTypes.shape({
    type: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    disabled: PropTypes.bool.isRequired,
    label: PropTypes.any.isRequired
  }).isRequired
};
