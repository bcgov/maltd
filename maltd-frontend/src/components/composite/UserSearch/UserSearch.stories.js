/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { storiesOf } from "@storybook/react";
import "bootstrap/dist/css/bootstrap.css";
import UserSearch from "./UserSearch";

const userSearch = {
  state: { isLoading: false }
};

const isLoadingUserSearch = {
  state: { isLoading: true }
};

const inputField = {
  name: "idir",
  type: "text",
  placeholder: "Enter IDIR username to find",
  valid: false,
  invalid: false,
  value: "",
  disabled: false
};

const generalButton = {
  type: "submit",
  color: "primary",
  disabled: true,
  block: false,
  active: false,
  outline: false,
  label: "Find",
  styling: "general-button align-right"
};

storiesOf("UserSearch", module)
  .add("default", () => (
    <UserSearch
      inputField={inputField}
      generalButton={generalButton}
      userSearch={userSearch}
    />
  ))
  .add("valid", () => (
    <UserSearch
      userSearch={userSearch}
      inputField={{ ...inputField, valid: true, value: "IDIR/validuser" }}
      generalButton={{ ...generalButton, disabled: false }}
    />
  ))
  .add("loading", () => (
    <UserSearch
      userSearch={isLoadingUserSearch}
      inputField={{
        ...inputField,
        valid: true,
        value: "IDIR/validuser",
        disabled: true
      }}
      generalButton={generalButton}
    />
  ))
  .add("invalid", () => (
    <UserSearch
      userSearch={userSearch}
      inputField={{ ...inputField, invalid: true, value: "a" }}
      generalButton={generalButton}
    />
  ));
