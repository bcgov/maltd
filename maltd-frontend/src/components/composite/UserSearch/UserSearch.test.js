import React from "react";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import UserSearch from "./UserSearch";

Enzyme.configure({ adapter: new Adapter() });

describe("User Search", () => {
  const inputField = {
    type: "text",
    name: "myinputfield",
    placeholder: "placeholder",
    valid: false,
    invalid: false,
    value: "idir",
    disabled: false
  };

  const generalButton = {
    type: "submit",
    color: "primary",
    disabled: true,
    label: "Find"
  };

  test("Component renders as expected", () => {
    const component = renderer.create(
      <UserSearch
        userSearch={{
          state: { isLoading: false, userExists: null }
        }}
        inputField={inputField}
        onClick={() => jest.fn()}
        onChange={() => jest.fn()}
        generalButton={generalButton}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Component does not render a loading message when no results are loading", () => {
    const component = shallow(
      <UserSearch
        userSearch={{
          state: { isLoading: false, userExists: null }
        }}
        inputField={inputField}
        onClick={() => jest.fn()}
        onChange={() => jest.fn()}
        generalButton={generalButton}
      />
    );

    expect(component.exists("#loading")).toEqual(false);
  });

  test("Component renders an error message when there are no results returned for a user search query", () => {
    const component = shallow(
      <UserSearch
        userSearch={{
          state: { isLoading: false, userExists: false }
        }}
        inputField={inputField}
        onClick={() => jest.fn()}
        onChange={() => jest.fn()}
        generalButton={generalButton}
      />
    );

    expect(component.exists(".error-message")).toEqual(true);
  });
});
