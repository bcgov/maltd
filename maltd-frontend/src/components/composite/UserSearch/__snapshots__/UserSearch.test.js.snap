// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Search Component renders as expected 1`] = `
<div>
  <p>
    Find a user by IDIR username and add them to or remove them from projects.
  </p>
  <p />
  <div>
    <input
      className="input-field form-control"
      defaultValue="idir"
      disabled={false}
      name="myinputfield"
      placeholder="placeholder"
      type="text"
    />
    <p />
    <button
      aria-label={null}
      className="general-button btn btn-primary disabled"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Find
    </button>
  </div>
</div>
`;
